### DISCUSSION TITLE
Implications of REDIS license change?

### DISCUSSION THREAD

**Discussion by fdrab** [StackStorm/st2#6172] [General]
Hello all,

As you've no doubt noticed, Redis is changing it's license to RSAL (similar to SSPL) and thus does (probably) no longer meets the definition of "Open Source": https://redis.io/docs/about/license/

Are there any implications on the ST2 project? Any alternatives / difficulty to swap?

**Comment by nzlosh** [comment]
Redis is dual licenced including SSPLv1, which is the same licence as MongoDB.  I'm not a lawyer, but my understanding is that StackStorm doesn't make Redis/MongoDB available as a service so section 13 isn't applicable.

Replacing MongoDB by another DB would be non-trivial.  Replacing Redis would be easier as the underlying `tooz` framework already has abstraction for multiple backends that provide distributed locking such as consul, etcd, zookeeper and others.

**Comment by ensignavenger** [comment]
As I am just starting to evaluate StackStorm for use in my company, the dependence on non-open-source components Redis and Mongo are definitely two huge negatives, and are making it more difficult to even evaluate it within my strictly open source environment.  My planned work around is to use one of the Redis forks and to try to find a "mongo-compatible" db that will work.  But that means I have to evaluate two different alternative databases and test them with StackStorm, before I can even begin a serious evaluation of StackStorm itself.

It would be great if the StackStorm community were to commit to supporting a specific, open source alternative to these products and add the alternatives to the testing to ensure ongoing compatibility.

I will try to report back here what products I select when I get that far, but I have several other projects on my plate right now so it may be a while before I get back to it, considering how much more work it will mean.

Thanks to everyone for creating and maintaining StackStorm, while I haven't gotten to use it yet it looks like a great solution in the space!

**Reply by fdrab** [reply]
You can probably use the community version of MongoDB (if you don't need enterprise-level features like encryption at rest) and you can swap REDIS with Zookeeper afaik.
I've tried documentDB as mongo replacement and it doesn't work (the instance installs, but any workflow execution fails).
